File: plugins/web/src/markdown.rs
//! Markdown → HTML for article bodies (GFM subset via pulldown-cmark).

use pulldown_cmark::{html, Options, Parser};

pub fn to_html(md: &str) -> String {
    let mut opts = Options::empty();
    opts.insert(
        Options::ENABLE_STRIKETHROUGH
            | Options::ENABLE_TABLES
            | Options::ENABLE_TASKLISTS,
    );
    let parser = Parser::new_ext(md, opts);
    let mut html_out = String::new();
    html::push_html(&mut html_out, parser);
    format!("<div class=\"article-body md\">{html_out}</div>")
}

File: plugins/web/internal/render/markdown.go
package render

import (
	"bytes"
	"strings"

	"github.com/yuin/goldmark"
	"github.com/yuin/goldmark/extension"
	"github.com/yuin/goldmark/parser"
	"github.com/yuin/goldmark/renderer/html"
)

// markdownToHTML renders article body Markdown to HTML (GFM subset), matching
// the Rust web plugin's pulldown-cmark path.
func markdownToHTML(md string) string {
	md = strings.TrimSpace(md)
	if md == "" {
		return ""
	}
	gm := goldmark.New(
		goldmark.WithExtensions(
			extension.GFM,
			extension.Strikethrough,
		),
		goldmark.WithParserOptions(
			parser.WithAutoHeadingID(),
		),
		goldmark.WithRendererOptions(
			html.WithHardWraps(),
			html.WithXHTML(),
		),
	)
	var buf bytes.Buffer
	if err := gm.Convert([]byte(md), &buf); err != nil {
		return `<div class="article-body"><p class="article-p">` + esc(md) + `</p></div>`
	}
	return `<div class="article-body md">` + buf.String() + `</div>`
}
